client/src/components/pages/sign-in/index.tsx

```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+import { Form, Formik } from 'formik';
+import { Link } from 'react-router-dom';
+import CoverLayout from './cover-layout';
+import PasswordField from './password-field';
+import FormField from './form-field';
+import Separator from './separator';
+import Button, { ButtonClasses } from './button';
+import { combineClasses } from 'helpers/combineClasses.helper';
+import styles from './sign-in.module.scss';
+
+function validateEmail(email: string): string | undefined {
+	if (!email) {
+		return 'Enter email';
+	} else if (!/^[a-z0-9._%+-]+@[a-z0-9.-]+\.[a-z]{2,4}$/i.test(email)) {
+		return 'Invalid email';
+	}
+}
+
+function validatePassword(password: string): string | undefined {
+	if (!password) {
+		return 'Enter password';
+	}
+}
```

Comment:
1. Better to use [Yup validation schema](https://formik.org/docs/guides/validation#validationschema) with formik. You will have existing solution for validating email and non-empty string.
2. Move it to `config.ts` in the same folder. sign-in/config.ts
3. Just having in mind. You do not need to use `else if` when you return some data in each `if` condition. It will work in the same way without `else`.